src/headless/CMakeLists.txt

```diff
@@ -0,0 +1,47 @@
+# vi:set sw=2 et:
+project(surge-headless)
+
+surge_add_lib_subdirectory(catch2)
+
+add_executable(${PROJECT_NAME}
+  HeadlessNonTestFunctions.cpp
+  HeadlessNonTestFunctions.h
+  HeadlessPluginLayerProxy.h
+  HeadlessUtils.cpp
+  HeadlessUtils.h
+  Player.cpp
+  Player.h
+  UnitTestUtilities.cpp
+  UnitTestUtilities.h
+  UnitTests.cpp
+  UnitTestsDSP.cpp
+  UnitTestsFLT.cpp
+  UnitTestsFX.cpp
+  UnitTestsINFRA.cpp
+  UnitTestsIO.cpp
+  UnitTestsLUA.cpp
+  UnitTestsMIDI.cpp
+  UnitTestsMOD.cpp
+  UnitTestsMSEG.cpp
+  UnitTestsPARAM.cpp
+  UnitTestsTUN.cpp
+  main.cpp
+)
+
+target_link_libraries(${PROJECT_NAME} PRIVATE
+  samplerate
+  surge::catch2
+  surge::surge-common
+)
+
+find_package(LibSndFile ${PACKAGE_OPTION})
```

Comment:
We only use libsndfile in one place so we can remove this and the cmake and when I need it again i can replace it with the juce wav support no problem. Lets clean this dependency up while we are doing this (the one place is in src/headless/ and its a test routine I don't use any more really)

Reply:
Great, this should also get rid of the scary "this libsndfile is linked against 10.15" warning on macOS.